docs/Concepts/Pruning.md

```diff
@@ -1,25 +1,61 @@
 ---
-description: Pruning
+description: Full and archive node types
 ---
 
-# Pruning
+# Full and archive node types
+
+Besu supports two node types, commonly referred to as _full nodes_ and _archive nodes_.
+
+Full nodes have just the current state of the blockchain so cannot serve the network with all data
+requests (for example, the balance of an account as at an old block). Full nodes can guarantee the
+latest state for the blockchain (and some older states, but not all). You can check current
+balances, sign and send transactions, and look at current Dapp data.
+
+Archive nodes have all of this and they also store the intermediary state of every account and
+contract for every block since the genesis block. With an archive node you can do everything you
+do with a full node, as well as access historical state data.
+
+Archive nodes require significantly more disk space (approximately 3TB) than full nodes
+(approximately 750GB).
+
+To run an archive node, enable full synchronization using
+[`--sync-mode=FULL`](../Reference/CLI/CLI-Syntax.md#sync-mode), which by default also disables
+pruning ([`--pruning-enabled=false`](../Reference/CLI/CLI-Syntax.md#pruning-enabled)).
+
+To run a full node, enable fast synchronization using
+[`--sync-mode=FAST`](../Reference/CLI/CLI-Syntax.md#sync-mode), which by default also enables
+pruning ([`--pruning-enabled=true`](../Reference/CLI/CLI-Syntax.md#pruning-enabled)).
+
+## Fast synchronization
+
+Instead of a full synchronization ([`--sync-mode=FULL`](../Reference/CLI/CLI-Syntax.md#sync-mode)),
+which starts from the genesis block and reprocesses all transactions, fast synchronization
+([`--sync-mode=FAST`](../Reference/CLI/CLI-Syntax.md#sync-mode)) downloads the transaction receipts
```

Comment:
Doesn't it also verify some blocks? I thought it did verify 1/n blocks, see: https://github.com/ethereum/go-ethereum/pull/1889. I'm not sure if our implementation is the same, though. cc: @RatanRSur 

Also, it's worth highlighting that past the pivot block, fast sync switches to full sync.

Reply:
It verifies the chain of headers from the genesis until a block when it switches to processing blocks in their entirety (full sync as you mentioned).

I would mention that it downloads the headers but I wouldn't mention that it switches to full sync at because it's an implementation detail and isn't really something they can rely on. The guarantee we give them is that they'll have the full current state but we don't want them assuming they'll have full state from any previous block because the pivot block switches many times while downloading the world state.

Reply:
Done.